backend/test/richard_burton_web/controllers/publication_controller_test.exs

```diff
@@ -623,4 +629,44 @@ defmodule RichardBurtonWeb.PublicationControllerTest do
       assert expected_data == response(conn, 200)
     end
   end
+
+  describe "update" do
+    def setup_update(%{conn: conn}) do
+      {:ok, publication} = @publication_attrs |> create_publications()
+      path = conn |> publication_path(:update, publication.id)
+
+      %{publication: publication, path: path}
+    end
+
+    # TODO:
+    #
+    # - tests de errores de validación
+    # - fallback controller para 404
+    #
+    setup [:mock_admin_logged_in, :setup_update]
+
+    test "returns 200 and updates the publication", %{
+      conn: conn,
+      path: path,
+      publication: publication
+    } do
+      update_attrs = %{
+        title: Faker.Lorem.sentence(),
+        year: (:rand.uniform() * 120 + 1900) |> trunc()
```

Comment:
Wondering if there's real benefit from using randomized input for the test. `Faker` may be fine to reduce the complexity of writing tests, though it adds an extra dependency. But the way `year` is generated here seems a bit complex for a 4 digit input. Please let me know if I'm missing something.